Documentation/proposals/202411-zone-aware-sharding.md

```diff
@@ -0,0 +1,227 @@
+# Status subresource for Prometheus operator CRDs
+
+* **Owners:**
+  * [arnecls](https://github.com/arnecls)
+
+* **Related Tickets:**
+  * [#6437](https://github.com/prometheus-operator/prometheus-operator/issues/6437)
+
+* **Other docs:**
+  * [Well known kubernetes labels](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone)
+  * [AWS zone names](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-availability-zones)
+  * [GCP zone names](https://cloud.google.com/compute/docs/regions-zones#available)
+
+This proposal describes how we can implement zone-aware sharding by adding
+support for custom labels and zone configuration options to the existing
+prometheus configuration resources. 
+
+## Why
+
+When running large, multi-zone clusters, prometheus scraping can lead to an
+increase in inter-zone traffic costs. The current sharding mechanics will
+allow multiple instances of prometheus to run, but as the `__address__` label
+is hard coded, all instances will always fetch all zones.
+
+It is not sufficient to simply switch this label with another label, as
+multiple prometheus instances per zone might be required. Furthermore we
+must be able to calculate the "assignment" of a specific instance to a zone,
+and that assignment must be stable.
+
+## Goals
+
+- Define a set of configuration options required to allow zone aware sharding
+- Define the relabel configuration to be generated for zone aware sharding
+- Define changes to the prometheus pod spec to support zone stickyness
+- Stay backwards compatible to the current mechanism by default
+
+## Non-goals
+
+- Implement mechanisms to avoid configuration errors by the user
+- Support mixed environments (kubernetes and non-kubernetes targets are scraped)
+
+## Algorithm
+
+In order to do calculate a stable assignment, following parameters are required:
+
+1. `num_shards`: The number of prometheus shards
+2. `shard_index`: A number of the range `[0..num_shards[` identifying a single prometheus 
+   instance
```

Comment:
(nit)
```suggestion
   instance inside a given shard
```

Reply:
I was under the impression that a „shard“ is an instance. Or is it a group of instances in the prometheus context?

Reply:
Ah - because a shard can also have multiple replica?